BigBaseV2/src/services/context_menu_service.cpp

```diff
@@ -69,6 +69,70 @@ namespace big
+		FUR.z = position.z + dimensions.y * forward_vector.z + dimensions.x * right_vector.z + dimensions.z * up_vector.z;
+
+		BLL.x = position.x - dimensions.y * forward_vector.x - dimensions.x * right_vector.x - dimensions.z * up_vector.x;
+		BLL.y = position.y - dimensions.y * forward_vector.y - dimensions.x * right_vector.y - dimensions.z * up_vector.y;
+		BLL.z = position.z - dimensions.y * forward_vector.z - dimensions.x * right_vector.z - dimensions.z * up_vector.z;
+
+		Vector3 edge1 = BLL;
+		Vector3 edge2;
+		Vector3 edge3;
+		Vector3 edge4;
+		Vector3 edge5 = FUR;
+		Vector3 edge6;
+		Vector3 edge7;
+		Vector3 edge8;
+
+		int green = colour * 255;
+		int blue = abs(colour - 1) * 255;
+
+		edge2.x = edge1.x + 2 * dimensions.y * forward_vector.x;
+		edge2.y = edge1.y + 2 * dimensions.y * forward_vector.y;
+		edge2.z = edge1.z + 2 * dimensions.y * forward_vector.z;
+
+		edge3.x = edge2.x + 2 * dimensions.z * up_vector.x;
+		edge3.y = edge2.y + 2 * dimensions.z * up_vector.y;
+		edge3.z = edge2.z + 2 * dimensions.z * up_vector.z;
+
+		edge4.x = edge1.x + 2 * dimensions.z * up_vector.x;
+		edge4.y = edge1.y + 2 * dimensions.z * up_vector.y;
+		edge4.z = edge1.z + 2 * dimensions.z * up_vector.z;
+
+		edge6.x = edge5.x - 2 * dimensions.y * forward_vector.x;
+		edge6.y = edge5.y - 2 * dimensions.y * forward_vector.y;
+		edge6.z = edge5.z - 2 * dimensions.y * forward_vector.z;
+
+		edge7.x = edge6.x - 2 * dimensions.z * up_vector.x;
+		edge7.y = edge6.y - 2 * dimensions.z * up_vector.y;
+		edge7.z = edge6.z - 2 * dimensions.z * up_vector.z;
+
+		edge8.x = edge5.x - 2 * dimensions.z * up_vector.x;
+		edge8.y = edge5.y - 2 * dimensions.z * up_vector.y;
+		edge8.z = edge5.z - 2 * dimensions.z * up_vector.z;
+
+
+		GRAPHICS::DRAW_LINE(edge1.x, edge1.y, edge1.z, edge2.x, edge2.y, edge2.z, 0, green, blue, 200);
+		GRAPHICS::DRAW_LINE(edge1.x, edge1.y, edge1.z, edge4.x, edge4.y, edge4.z, 0, green, blue, 200);
+		GRAPHICS::DRAW_LINE(edge2.x, edge2.y, edge2.z, edge3.x, edge3.y, edge3.z, 0, green, blue, 200);
+		GRAPHICS::DRAW_LINE(edge3.x, edge3.y, edge3.z, edge4.x, edge4.y, edge4.z, 0, green, blue, 200);
+
+		GRAPHICS::DRAW_LINE(edge5.x, edge5.y, edge5.z, edge6.x, edge6.y, edge6.z, 0, green, blue, 200);
+		GRAPHICS::DRAW_LINE(edge5.x, edge5.y, edge5.z, edge8.x, edge8.y, edge8.z, 0, green, blue, 200);
+		GRAPHICS::DRAW_LINE(edge6.x, edge6.y, edge6.z, edge7.x, edge7.y, edge7.z, 0, green, blue, 200);
+		GRAPHICS::DRAW_LINE(edge7.x, edge7.y, edge7.z, edge8.x, edge8.y, edge8.z, 0, green, blue, 200);
+
+		GRAPHICS::DRAW_LINE(edge1.x, edge1.y, edge1.z, edge7.x, edge7.y, edge7.z, 0, green, blue, 200);
+		GRAPHICS::DRAW_LINE(edge2.x, edge2.y, edge2.z, edge8.x, edge8.y, edge8.z, 0, green, blue, 200);
+		GRAPHICS::DRAW_LINE(edge3.x, edge3.y, edge3.z, edge5.x, edge5.y, edge5.z, 0, green, blue, 200);
+		GRAPHICS::DRAW_LINE(edge4.x, edge4.y, edge4.z, edge6.x, edge6.y, edge6.z, 0, green, blue, 200);
```

Comment:
We don't use natives to draw anything in-game, please use ImGui instead.

You can take a look at the ESP how it is made https://github.com/Yimura/YimMenu/blob/master/BigBaseV2/src/views/esp/view_esp.cpp

The context menu also has a UI function so if you need to draw anything you can do it there.
https://github.com/Yimura/YimMenu/blob/master/BigBaseV2/src/views/view_context_menu.cpp